### PR TITLE
CATROID-271 GSoC refactor PreferencesSensorMappingTest

### PR DESCRIPTION
**Pull request by gPathpp**

### CODE REVIEW

File: catroid/src/androidTest/java/org/catrobat/catroid/test/devices/mindstorms/nxt/PreferencesSensorMappingTest.java [hunk 1]
```diff
@@ -25,45 +25,74 @@
...
-		for (int i = 0; i < correctMapping.size(); ++i) {
-			assertEquals(context.getString(correctMapping.get(i).first), sensorNames[i]);
+	private ArrayList<String> sensorNames = new ArrayList<>();
+	private ArrayList<String> sensorCodes = new ArrayList<>();
 
-			assertEquals(correctMapping.get(i).second.getSensorCode(), sensorPreferencesCodes[i]);
-		}
+	@Before
+	public void setUp() {
+		Context context = InstrumentationRegistry.getTargetContext().getApplicationContext();
```
Comment (DinosaurierRex) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/devices/mindstorms/nxt/PreferencesSensorMappingTest.java]: `...getTargetContext();` should be enough to return the currents instrumented applications contex

**Comment by wslany** [comment]
Thanks!

File: catroid/src/androidTest/java/org/catrobat/catroid/test/devices/mindstorms/nxt/PreferencesSensorMappingTest.java [hunk 2]
```diff
@@ -25,45 +25,74 @@
...
-		assertEquals(correctMapping.size(), sensorPreferencesCodes.length);
+	private ArrayList<String> sensorNames = new ArrayList<>();
+	private ArrayList<String> sensorCodes = new ArrayList<>();
 
-		for (int i = 0; i < correctMapping.size(); ++i) {
-			assertEquals(context.getString(correctMapping.get(i).first), sensorNames[i]);
+	@Before
+	public void setUp() {
+		Context context = InstrumentationRegistry.getTargetContext();
+		sensorNames.addAll(Arrays.asList(context.getResources().getStringArray(R.array.nxt_sensor_chooser)));
```
Comment (84n4n4) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/devices/mindstorms/nxt/PreferencesSensorMappingTest.java]: simple add is sufficient

File: catroid/src/androidTest/java/org/catrobat/catroid/test/devices/mindstorms/nxt/PreferencesSensorMappingTest.java [hunk 3]
```diff
@@ -25,45 +25,74 @@
...
+	@Test
+	public void testNameAndPreferenceSameIndex() {
+		int sensorNameIndex = sensorNames.indexOf(sensorName);
+		int sensorCodeIndex = sensorCodes.indexOf(sensor.getSensorCode());
+		assertSame(sensorCodeIndex, sensorNameIndex);
+	}
+
+	@Test
+	public void testMappingContainsNameAndCode() {
+		assertTrue(sensorNames.contains(sensorName));
```
Comment (84n4n4) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/devices/mindstorms/nxt/PreferencesSensorMappingTest.java]: assertThat(sensorNames, hasItem(sensorName))

File: catroid/src/androidTest/java/org/catrobat/catroid/test/devices/mindstorms/nxt/PreferencesSensorMappingTest.java [hunk 4]
```diff
@@ -25,45 +25,74 @@
...
+		assertTrue(sensorCodes.contains(sensor.getSensorCode()));
+	}
+
+	@Test
+	public void testMappingSize() {
+		assertSame(EXPECTED_MAPPING_SIZE, sensorNames.size());
```
Comment (84n4n4) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/devices/mindstorms/nxt/PreferencesSensorMappingTest.java]: assertEquals!!!

**Pull request merged**